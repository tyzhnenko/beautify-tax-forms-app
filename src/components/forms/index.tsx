import F1401803 from "./F1401803/DFSForm";
import F1419101 from "./F1419101/DFSForm";
import F1419103 from "./F1419103/DFSForm";
import F1419104 from "./F1419104/DFSForm";

export const availableForms = [
  {
    name: "F1401803",
    component: F1401803,
    description:
      "Відомості з Державного реєстру фізичних осіб – платників податків про суми виплачених доходів та утриманих податків. Відповідь на запит у електронному вигляді",
  },
  {
    name: "F1419101",
    component: F1419101,
    description:
      "Вiдомостi з Державного реєстру фiзичних осiб – платникiв податкiв про джерела та суми нарахованого доходу, нарахованого (перерахованого) податку та вiйськового збору (вiдповiдь на запит в електронному виглядi) (v1)",
  },
  {
    name: "F1419103",
    component: F1419103,
    description:
      "Вiдомостi з Державного реєстру фiзичних осiб – платникiв податкiв про джерела та суми нарахованого доходу, нарахованого (перерахованого) податку та вiйськового збору (вiдповiдь на запит в електронному виглядi) (v3)",
  },
  {
    name: "F1419104",
    component: F1419104,
    description:
      "Вiдомостi з Державного реєстру фiзичних осiб – платникiв податкiв про джерела та суми нарахованого доходу, нарахованого (перерахованого) податку та вiйськового збору (вiдповiдь на запит в електронному виглядi) (v4)",
  },
];
