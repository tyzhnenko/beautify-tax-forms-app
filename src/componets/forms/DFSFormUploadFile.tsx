import {
  Button,
  Dialog,
  DialogHeader,
  DialogBody,
  DialogFooter,
  Input,
} from "@material-tailwind/react";
import { Fragment, useState } from "react";
import readXMLFile from "../../utils/readXMLFile";

function DFSFormUploadFile(props: any) {
  const [open, setOpen] = useState(false);
  const [file, setFile] = useState<File>();

  function onChangeHandler(event: React.ChangeEvent<HTMLInputElement>) {
    if (!event.target.files) {
      return;
    }
    setFile(event.target.files[0]);
  }

  const openHandler = () => setOpen(!open);

  function uploadClickHandler() {
    if (!file) {
      openHandler();
      return;
    }

    readXMLFile(file, props.xmlLoader);
    setFile(undefined);
    openHandler();
  }

  return (
    <div className="text-center">
      <Fragment>
        <Button onClick={openHandler}>Оберить файл</Button>
        <Dialog open={open} handler={openHandler}>
          <DialogHeader>Вибір файлу</DialogHeader>
          <DialogBody>
            Оберить XML файл форми F0121213 для відображення
            <Input
              data-testid="file-selector"
              type="file"
              onChange={onChangeHandler}
              accept="text/xml"
            />
          </DialogBody>
          <DialogFooter>
            <Button variant="text" color="red" onClick={openHandler}>
              Відмінити
            </Button>
            <Button
              variant="gradient"
              color="green"
              onClick={uploadClickHandler}
            >
              Обробити
            </Button>
          </DialogFooter>
        </Dialog>
      </Fragment>
    </div>
  );
}

export default DFSFormUploadFile;
