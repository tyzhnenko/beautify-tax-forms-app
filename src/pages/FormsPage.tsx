import { Link } from "react-router-dom";
import { availableForms } from "../components/forms";
import useTitle from "../utils/useTitle";

function FormsPage() {
  useTitle("Доступні форми");

  return (
    <div className="pt-10">
      <h1 className="text-4xl text-center pb-10">Доступні форми</h1>
      {availableForms.map((form) => (
        <div className="grid grid-cols-12" key={form.name}>
          <div className="col-span-3 text-center">
            <Link to={`/forms/${form.name}`}>{form.name}</Link>
          </div>
          <div className="col-span-9">{form.description}</div>
        </div>
      ))}
    </div>
  );
}

export default FormsPage;
