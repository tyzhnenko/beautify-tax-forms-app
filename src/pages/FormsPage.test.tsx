/**
 * @jest-environment jsdom
 */

import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";
import FormsPage from "./FormsPage";
import { availableForms } from "../components/forms";

describe("AppRouter component", () => {
  it("should render the component", () => {
    render(
      <BrowserRouter>
        <FormsPage />
      </BrowserRouter>
    );
    expect(document.title).toEqual("Доступні форми");
    availableForms.forEach((form) => {
      expect(screen.getByText(form.name)).toBeInTheDocument();
      expect(screen.getByText(form.description)).toBeInTheDocument();
      expect(screen.getByRole("link", { name: form.name })).toHaveAttribute(
        "href",
        `/forms/${form.name}`
      );
    });
  });

  it("should match snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <FormsPage />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
