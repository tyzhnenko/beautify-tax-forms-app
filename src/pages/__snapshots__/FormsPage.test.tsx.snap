// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter component should match snapshot 1`] = `
<div>
  <div
    class="pt-10"
  >
    <h1
      class="text-4xl text-center pb-10"
    >
      Доступні форми
    </h1>
    <div
      class="grid grid-cols-12"
    >
      <div
        class="col-span-3 text-center underline"
      >
        <a
          href="/forms/F1401803"
        >
          F1401803
        </a>
      </div>
      <div
        class="col-span-9"
      >
        Відомості з Державного реєстру фізичних осіб – платників податків про суми виплачених доходів та утриманих податків. Відповідь на запит у електронному вигляді
      </div>
    </div>
    <div
      class="grid grid-cols-12"
    >
      <div
        class="col-span-3 text-center underline"
      >
        <a
          href="/forms/F1419101"
        >
          F1419101
        </a>
      </div>
      <div
        class="col-span-9"
      >
        Вiдомостi з Державного реєстру фiзичних осiб – платникiв податкiв про джерела та суми нарахованого доходу, нарахованого (перерахованого) податку та вiйськового збору (вiдповiдь на запит в електронному виглядi) (v1)
      </div>
    </div>
    <div
      class="grid grid-cols-12"
    >
      <div
        class="col-span-3 text-center underline"
      >
        <a
          href="/forms/F1419103"
        >
          F1419103
        </a>
      </div>
      <div
        class="col-span-9"
      >
        Вiдомостi з Державного реєстру фiзичних осiб – платникiв податкiв про джерела та суми нарахованого доходу, нарахованого (перерахованого) податку та вiйськового збору (вiдповiдь на запит в електронному виглядi) (v3)
      </div>
    </div>
    <div
      class="grid grid-cols-12"
    >
      <div
        class="col-span-3 text-center underline"
      >
        <a
          href="/forms/F1419104"
        >
          F1419104
        </a>
      </div>
      <div
        class="col-span-9"
      >
        Вiдомостi з Державного реєстру фiзичних осiб – платникiв податкiв про джерела та суми нарахованого доходу, нарахованого (перерахованого) податку та вiйськового збору (вiдповiдь на запит в електронному виглядi) (v4)
      </div>
    </div>
  </div>
</div>
`;
